### PR TITLE
Long table name error hotfix

### PR DESCRIPTION
**Pull request by MaximMoinat**
Solution for issues #14 and #113, where long table names  causes errors in WR scans.

This fix prevents RiaH from crashing by skipping the second table when a tab name clash occurs. It presents user with a warning when this happens.

### CODE REVIEW

File: whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java [hunk 1]
```diff
@@ -1,484 +1,499 @@
...
+							Long.valueOf(fieldInfo.nProcessed), fieldInfo.getFractionEmpty());
+				addRow(sheet, "");
+			}
+
+			// Create per table sheets
+			for (String table : tables) {
+				String sheetName = createSheetName(table);
+
+				if (workbook.getSheet(sheetName) != null) {
+					System.out.println("Worksheet already exists in the scan report, the current table is skipped: " + sheetName);
```
Comment (blootsvoets) [inline whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java]: This should probably be a UI message? Suggested alternative, add a `tries` parameter to `createSheetName`:
```java
String sheetName = null;
int tries = 0;
while (sheetName == null && tries < 10) {
  sheetName = createSheetName(table, tries);
  if (workbook.getSheet(sheetName) != null) {
    sheetName = null;
  }
  tries++;
}
if (sheetName == null) {
// throw some warning
}

String createSheetName(String tableName, int tries) {
  String name = tableName;
  if (name.length() > 29) {
    name = name.substring(0, 29);
  }
  if (tries > 0) {
    name += "_" + tries;
  }
  return tries;
}
```

Comment (MaximMoinat) [reply whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java]: Implemented a more rigorous, less elegant, approach and prepended a sheet index to all long names. This allows RiaH to get the corresponding sheet to load value counts, with backwards compatibility to older scan reports. See commit [add9f42](https://github.com/OHDSI/WhiteRabbit/pull/162/commits/add9f42481fc6b3120fbe8461bfb5d5b3a956307).

The `System.out.println` is redirected to a messages box in the UI.

File: whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java [hunk 2]
```diff
@@ -117,28 +118,43 @@ private void generateReport(Map<String, List<FieldInfo>> tableToFieldInfos, Stri
...
+			int sheetIndex = 0;
+			Map<String, String> sheetNameLookup = new HashMap<>();
+			for (String tableName : tables) {
+				// Create unique sheetNames from table name and index.
+				String sheetName = Table.createSheetNameFromTableName(tableName, sheetIndex);
+				sheetNameLookup.put(tableName, sheetName);
+
+				String tableNameIndexed = tableName;
+				if (tableName.length() > 31) {
+					tableNameIndexed = Integer.toString(sheetIndex) + '_' + tableName;
```
Comment (blootsvoets) [inline whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java]: Is this reversal compared to `createShssetNameFromTableName` intentional?

Comment (blootsvoets) [reply whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java]: Never mind, the index + sheet name order is the same.

Comment (MaximMoinat) [reply whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java]: Just separated this a bit better from the main code.

**Comment by MaximMoinat** [comment]
One thing that is mentioned in #113 is to remove the '.csv.' extension from filenames. While I think this is in general useful, it could create clashes in file name uniqueness (e.g. same filename, but with .csv and .txt extension). This then again creates similar problems with RiaH not being able to find the field values.
Bottomline, lets not add that feature.

**Pull request merged**